app/models/nic/interface.rb

```diff
@@ -65,12 +65,12 @@ def normalize_name
       # Remove whitespace
       self.name.gsub!(/\s/,'')
 
-      if domain.nil? and name.match(/\./)
+      if domain.nil? && name.match(/\./) && !changed_attributes['domain_id'].present?
```

Comment:
I think this methods needs clearing, after many minutes of thinking I came up with the meaning of these conditions

```
if interface domain_id is wrong or domain_id is nil && we have domain name in name && this interfaces is new record or the interface previously wasn't assigned domain
  we try to match the Domain based on name and assign it
else
  if this is existing interface && the domain was already set to something (now we could have removed it or changed it to another id)
    we find old_domain
    extract old_domain name from current name
  end
  set name to fqdn regardles of ^
end
```

Does this make sense? Not to me, that's for sure... Since I don't see any unit test for this method, I would prefer first covering it by tests, then trying to refactor and then do the changes/fixes (I'm OK with doing this in one PR)